CONTRIBUTING.md

```diff
@@ -0,0 +1,79 @@
+# Contributing to ISOBUS++
+
+We accept public contributions that follow our code of conduct, pass all automated pre-merge checks, and pass a manual code review by a repo maintainer.
+
+## What Are The Requirements?
+
+Our style rules are based loosely on Autosar and MISRA suggestions and try to ensure the highest quality possible.
+
+* Contributions must follow the style defined in our `.clang-format` file. You can ensure you pass this check by running `find . -iname *.hpp -o -iname *.cpp | xargs clang-format -i` at the root of the repo.
+* Contributions should follow these additional style requirements, which will be check in code reviews.
+	* Function names `snake_case`
+	* Constant values `CAPITALIZED_SNAKE`
+	* Constants on the left in `==` and `!=` checks. Like this: `if (5 == value)` NOT `if (value == 5)`. This is to prevent accidentally omitting an `=` in the operator and creating a runtime bug.
+* Doxygen should compile with no warnings when run at the root of the project `doxygen doxyfile`
+* Copyright notice must be included in each file
+* Builds must pass the compilation github action
+* `NULL` should not be used when `nullptr` can be used
+* Prefer C++ over C
+* C++11 or earlier is required to help provide the maximum compatibility to all modern compilers, even for embedded ones that do not support the most up-to-date versions of the standard, such as ARM Compiler 5 and 6.
+
+## Setting up a Development Environment With GUI
+
+You can easily set up a linux PC or virutal machine to develop for this project!
+
+We suggest using Ubuntu or Raspian, but other systems may also work. Windows development is supported with PEAK CAN hardware devices.
+
+### Install The Prerequisites
+
+Linux:
+
+Install git, a working CXX compiler, cmake, and git
```

Comment:
git is written here twice, could replace with "Install the following tools:" to avoid maintenance of writing dependencies multiple times?

Reply:
Changed to list of tools, with some marked as optional